superset/assets/src/dashboard/components/FilterIndicatorsContainer.jsx

```diff
@@ -0,0 +1,162 @@
+          directPathToFilter,
+          scope,
+          isDateFilter,
+          isInstantFilter,
+          columns,
+        } = dashboardFilter;
+
+        // do not apply filter on filter_box itself
+        // do not apply filter on filter_immune_slices list
+        if (
+          currentChartId !== chartId &&
+          !filter_immune_slices.includes(currentChartId)
+        ) {
+          Object.keys(columns).forEach(name => {
+            const colorMapKey = getFilterColorKey(chartId, name);
+            const directPathToLabel = directPathToFilter.slice();
+            directPathToLabel.push(`LABEL-${name}`);
+            const indicator = {
+              chartId,
+              colorCode: dashboardFiltersColorMap[colorMapKey],
+              componentId,
+              directPathToFilter: directPathToLabel,
+              scope,
+              isDateFilter,
+              isInstantFilter,
+              name,
+              values: columns[name],
+            };
+
+            // do not apply filter on fields in the filter_immune_slice_fields map
+            if (
+              filter_immune_slice_fields[currentChartId] &&
+              filter_immune_slice_fields[currentChartId].includes(name)
+            ) {
+              return;
+            }
+
+            if (isEmptyValue(indicator.values)) {
+              indicators[1].push(indicator);
+            } else {
+              indicators[0].push(indicator);
+            }
+          });
+        }
+
+        return indicators;
+      },
+      [[], []],
+    );
+
+    // cypress' electron don't support [].flat():
+    return [...sorteIndicatorsByEmptiness[0], ...sorteIndicatorsByEmptiness[1]];
+  }
+
+  render() {
+    const { chartStatus, setDirectPathToChild } = this.props;
+    if (chartStatus === 'loading') {
+      return null;
+    }
+
+    const indicators = this.getFilterIndicators();
+    // if total indicators <= 5, show all
+    // else: show top 4 indicators, and show all the rest in group
+    const showExtraIndicatorsInGroup =
+      indicators.length > FILTER_INDICATORS_DISPLAY_LENGTH + 1;
+
+    return (
+      <div className="dashboard-filter-indicators-container">
+        {indicators
+          .filter((indicator, index) =>
+            showExtraIndicatorsInGroup
+              ? index < FILTER_INDICATORS_DISPLAY_LENGTH
+              : true,
+          )
+          .map((indicator, index) => (
+            <FilterIndicator
+              key={index}
+              indicator={indicator}
+              setDirectPathToChild={setDirectPathToChild}
+            />
+          ))}
+        {showExtraIndicatorsInGroup && (
+          <FilterIndicatorGroup
+            indicators={indicators.slice(FILTER_INDICATORS_DISPLAY_LENGTH)}
+            setDirectPathToChild={setDirectPathToChild}
+          />
+        )}
+      </div>
+    );
+  }
+}
+
+FilterIndicatorsContainer.propTypes = propTypes;
+FilterIndicatorsContainer.defaultProps = defaultProps;
```

Comment:
don't you need to export after this? if not, i think it's the standard order in most of these files